File: ios/SuperCameraKit/Sources/SuperCameraKit/SessionMetrics.swift
import Foundation

public struct SessionMetrics {
    public var firstFrameLatencyMs: Int = 0
    public var previewFps: Int = 0
    public var renderDropRatePermille: Int = 0
    public var encoderInputFps: Int = 0
    public var avDriftMs: Int = 0
    public var modeSwitchLatencyMs: Int = 0

    public init() {}
}

File: android/supercamera/src/main/java/com/dcamera/telemetry/SessionMetrics.kt
package com.dcamera.telemetry

data class SessionMetrics(
    var firstFrameLatencyMs: Int = 0,
    var previewFps: Int = 0,
    var renderDropRatePermille: Int = 0,
    var encoderInputFps: Int = 0,
    var avDriftMs: Int = 0,
    var modeSwitchLatencyMs: Int = 0
)
